lib/modules/manager/kustomize/artifacts.ts

```diff
@@ -0,0 +1,237 @@
+  const pathExists = await localPathExists(path);
+
+  return pathExists ? path : null;
+}
+
+function helmRepositoryArgs(
+  repository: string,
+  depName: string,
+  datasource?: string,
+): string {
+  switch (datasource) {
+    case HelmDatasource.id:
+      return `--repo ${quote(repository)} ${depName}`;
+    case DockerDatasource.id:
+      return quote(`oci://${repository}`);
+    /* v8 ignore next 2: should never happen */
+    default:
+      throw new Error(`Unknown datasource: ${datasource}`);
+  }
+}
+
+async function inflateHelmChart(
+  flagEnabled: boolean,
+  execOptions: ExecOptions,
+  chartHome: string,
+  depName: string,
+  repository: string,
+  currentVersion: string,
+  newVersion?: string,
+  datasource?: string,
+): Promise<void> {
+  const currentChartExistingPath = await localExistingChartPath(
+    chartHome,
+    depName,
+    currentVersion,
+  );
+
+  if (!flagEnabled && is.nullOrUndefined(currentChartExistingPath)) {
+    logger.debug(
+      `Not inflating Helm chart for ${depName} as kustomizeInflateHelmCharts is not enabled and the current version isn't inflated`,
+    );
+    return;
+  }
+
+  if (
+    is.nonEmptyString(currentChartExistingPath) &&
+    is.nonEmptyString(newVersion)
+  ) {
+    logger.debug(`Deleting previous helm chart: ${currentChartExistingPath}`);
+    await deleteLocalFile(currentChartExistingPath);
+  }
+
+  const versionToPull = newVersion ?? currentVersion;
+  const versionToPullExistingPath = await localExistingChartPath(
+    chartHome,
+    depName,
+    versionToPull,
+  );
+
+  if (is.nonEmptyString(versionToPullExistingPath)) {
+    logger.debug(
+      `Helm chart ${depName} version ${versionToPull} already exists at ${versionToPullExistingPath}`,
+    );
+    return;
+  }
+
+  const folderName = `${depName}-${versionToPull}`;
+  const untarDir = upath.join(chartHome, folderName);
+  logger.debug(
+    `Pulling helm chart ${depName} version ${versionToPull} to ${untarDir}`,
+  );
+
+  const cmd =
+    `helm pull --untar --untardir ${quote(untarDir)} ` +
+    `--version ${quote(versionToPull)} ${helmRepositoryArgs(repository, depName, datasource)}`;
+
+  await exec(cmd, execOptions);
+}
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.debug(`kustomize.updateArtifacts(${packageFileName})`);
+  const project = parseKustomize(newPackageFileContent);
+  const isUpdateOptionInflateChartArchives =
+    config.postUpdateOptions?.includes('kustomizeInflateHelmCharts') === true;
+  if (is.nullOrUndefined(project)) {
+    return [
+      {
+        artifactError: {
+          stderr: 'Failed to parse new package file content',
+        },
+      },
+    ];
+  }
+
+  const chartHome = getSiblingFileName(
+    packageFileName,
+    project.helmGlobals?.chartHome as string, // eslint-disable-line @typescript-eslint/non-nullable-type-assertion-style
```

Comment:
`lint` complained about `@typescript-eslint/non-nullable-type-assertion-style` and `lint-fix` changed this to:
```typescript
project?.helmGlobals?.chartHome!
```
`lint` then complained about `@typescript-eslint/no-non-null-asserted-optional-chain`.

`chartHome` always has a value because we set the kustomize default if it's not explicitly set, and it wasn't optional in https://github.com/renovatebot/renovate/pull/34277/commits/c0761652c042e31124a2f4abbfe6870241a8cdfe but Michael wanted it changed in his comment [here](https://github.com/renovatebot/renovate/pull/34277/commits/c0761652c042e31124a2f4abbfe6870241a8cdfe#r1959600370).

I don't know what else to do but disable the lint at this point...

Reply:
```suggestion
    project.helmGlobals!.chartHome!
```
or 
```suggestion
    project.helmGlobals?.chartHome ?? 'charts';
```